File: twitter-clone-backend/src/main/java/com/mohamed/halim/twitterclone/model/Media.java
package com.mohamed.halim.twitterclone.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder(toBuilder = true)
@Entity
public class Media {
    @Id
    private Long id;
    @Enumerated(EnumType.STRING)
    private MediaType type;
    private String url;
    private Long duration;
    private int height;
    private int width;
    
    
}

File: twitter-clone-frontend/src/model/Media.ts
export default interface Media {
    id: number;
    type: "IMAGE"|"VIDEO";
    url: string;
    duration: number;
    height: number;
    width: number;
  }


  
 export function isMedia(obj: any): obj is Media {
    const keys: (keyof Media)[] = ['id', 'type', 'url', 'duration', 'height', 'width'];
    for (const key of keys) {
      if (!(key in obj)) {
        return false;
      }
    }
    return true;
  }
  
  